module/VuFind/tests/unit-tests/src/VuFindTest/RecordTab/VersionsTest.php

```diff
@@ -0,0 +1,130 @@
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+namespace VuFindTest\RecordTab;
+
+use Laminas\Config\Config;
+use VuFind\RecordTab\Versions;
+
+/**
+ * Versions Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class VersionsTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test getting Description.
+     *
+     * @return void
+     */
+    public function testGetDescription(): void
+    {
+        $count=5;
+        $som = $this->getMockPluginManager();
+        $config = $this->getMockConfig();
+        $recordDriver = $this->getMockBuilder(\VuFind\RecordDriver\SolrDefault::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $recordDriver->expects($this->any())->method('tryMethod')
+            ->with($this->equalTo('getOtherVersionCount'))
+            ->will($this->returnValue($count));
+        $obj= new Versions($config, $som);
+        $obj->setRecordDriver($recordDriver);
+        $translator = $this->getMockBuilder(\Laminas\I18n\Translator\TranslatorInterface::class)
+            ->getMock();
+        $translator->expects($this->any())->method('translate')
+            ->with($this->equalTo('other_versions_title'), $this->equalTo('default'))
+            ->will($this->returnValue("Count:%%count%%"));
+        $obj->setTranslator($translator);
+        $obj->getDescription();
+        $this->assertEquals("Count:5", $obj->getDescription());
+    }
+
+    /**
+     * Test if the tab is active.
+     *
+     * @return void
+     */
+    public function testisActive(): void
+    {
+        $som = $this->getMockPluginManager();
+        $config = $this->getMockConfig();
+        $optionsMock = $this->getMockBuilder(\VuFind\Search\Base\Options::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $som->expects($this->any())->method('get')
+            ->with($this->equalTo('foo'))
+            ->will($this->returnValue($optionsMock));
+        $optionsMock->expects($this->any())->method('getVersionsAction')
+            ->willReturnOnConsecutiveCalls(true, false);
+        $recordDriver = $this->getMockBuilder(\VuFind\RecordDriver\SolrDefault::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $recordDriver->expects($this->any())->method('getSourceIdentifier')
+            ->will($this->returnValue('foo'));
+        $recordDriver->expects($this->any())->method('tryMethod')
+            ->with($this->equalTo('getOtherVersionCount'))
+            ->willReturnOnConsecutiveCalls(1, 0);
+        $obj= new Versions($config, $som);
+        $obj->setRecordDriver($recordDriver);
+        //Test when the tab is active
+        $this->assertTrue($obj->isActive());
```

Comment:
If you rewrite this to use a dataProvider, I think you can tighten things up and simplify code a bit -- instead of using many $this->any assertions and the willReturnOnConsecutiveCalls logic, you could just use $this->once() for everything and return a mock value injected as a parameter by the data provider.